Lib/test/test_argparse.py

```diff
@@ -2836,6 +2836,29 @@ def test_help(self):
               '''
         self.assertEqual(parser.format_help(), textwrap.dedent(expected))
 
+    def test_optional_order(self):
+        parser = ErrorRaisingArgumentParser(prog="PROG")
+        group = parser.add_mutually_exclusive_group(required=True)
+        group.add_argument("-", dest="FOO")
```

Comment:
"-" is a very uncommon case. I think that you can test this with '--foo'.

BTW, use single quotes, they are more common in this file.